File: Desafio-Front/src/types/categoryResponse.ts
export enum CategoryType
{
  AMBAS = "AMBAS",
  DESPESA = "DESPESA",
  RECEITA = "RECEITA"
}

export type CategoryResponse = {
  id: string
  description: string
  categoryType: CategoryType
  isActive: boolean
}

export type ApiCategoryResponse = {
  success: boolean
  message: string
  data?: CategoryResponse
}

File: Desafio-WebAPI/Desafio-WebAPI/Models/Response/CategoryResponse.cs
using Desafio_WebAPI.Models.Entities;

namespace Desafio_WebAPI.Models.Response;

public class CategoryResponse
{
    public Guid Id { get; set; }
    public string Description { get; set; } = null!;
    public CategoryType CategoryType { get; set; }
    public bool IsActive { get; set; }
}
